api/handlers/admin.go

```diff
@@ -0,0 +1,199 @@
+package handlers
+
+import (
+	"blogs/api/services"
+	"blogs/common/helpers"
+	"blogs/pkg/loggers"
+	"blogs/pkg/models"
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+)
+
+type AdminHandler struct {
+	services.AdminServices
+}
+
+// retrieve every users records
+func (handler *AdminHandler) GetAllUsers(ctx echo.Context) error {
+	var users []models.Users
+	var limit, offset int
+
+	param := ctx.QueryParam("limit")
+	if param == "" {
+		limit = 100
+	} else {
+		convLimit, err := strconv.Atoi(param)
+		if err != nil {
+			loggers.WarningLog.Println(err)
+			return ctx.JSON(http.StatusBadRequest, helpers.ResponseJson{
+				Error: err.Error(),
+			})
+		}
+		limit = convLimit
+	}
+
+	param = ctx.QueryParam("offset")
+	if param == "" {
+		offset = 0
+	} else {
+		convOffset, err := strconv.Atoi(param)
+		if err != nil {
+			loggers.WarningLog.Println(err)
+			return ctx.JSON(http.StatusBadRequest, helpers.ResponseJson{
+				Error: err.Error(),
+			})
+		}
+		offset = convOffset
+	}
+
+	getRole := (ctx.Get("role"))
+	role := getRole.(string)
+
+	//call the get Users service
+	status, err := handler.AdminServices.GetUsers(&users, role, limit, offset)
+	if err != nil {
+		loggers.WarningLog.Println(err)
+		return ctx.JSON(status, helpers.ResponseJson{
+			Error: err.Error(),
+		})
+	}
+
+	return ctx.JSON(status, helpers.ResponseJson{
+		Message: "Users retreived successfully",
+		Data:    users,
+	})
+}
+
+// retrieve a single user record
+func (handler *AdminHandler) GetSingleUser(ctx echo.Context) error {
```

Comment:
Use proper function names if you get all users the function name should be GetUsers if it is single user it should be GetUser